src/NCronJob/JobExecutionContext.cs

```diff
@@ -6,7 +6,20 @@ internal sealed record JobExecutionContext : IJobExecutionContext
 {
     internal bool ExecuteChildren = true;
 
-    public JobExecutionContext(JobRun jobRun) => JobRun = jobRun;
+    public JobExecutionContext(JobRun jobRun)
+    {
+        JobRun = jobRun;
+        if (jobRun.JobDefinition.IsStartupJob)
```

Comment:
A cron job can also be defined as a startup job, so this may not be working either if we only consider the JobDefinition

Reply:
Yeah you are absolutely right - and I you convinced me that it isn't as straightforward as I thought. So I'll close this for now and have to rethink